resources/todomvc/vanilla-examples/javascript-web-components/src/components/todo-item/todo-item.component.js

```diff
@@ -0,0 +1,181 @@
+import template from "./todo-item.template.js";
+import { useDoubleClick } from "../../hooks/useDoubleClick.js";
+import { useKeyListener } from "../../hooks/useKeyListener.js";
+
+import globalStyles from "../../styles/global.constructable.js";
+import itemStyles from "../../styles/todo-item.constructable.js";
+
+class TodoItem extends HTMLElement {
+    static get observedAttributes() {
+        return ["id", "title", "completed"];
+    }
+
+    constructor() {
+        super();
+
+        this.id = "";
+        this.title = "Todo Item";
+        this.completed = "false";
+
+        const node = document.importNode(template.content, true);
+        this.item = node.querySelector(".todo-item");
+        this.toggleLabel = node.querySelector(".toggle-todo-label");
+        this.toggleInput = node.querySelector(".toggle-todo-input");
+        this.todoText = node.querySelector(".todo-item-text");
+        this.todoButton = node.querySelector(".remove-todo-button");
+        this.editLabel = node.querySelector(".edit-todo-label");
+        this.editInput = node.querySelector(".edit-todo-input");
+
+        this.shadow = this.attachShadow({ mode: "open" });
+        this.htmlDirection = document.querySelector("html").getAttribute("dir") || "ltr";
+        this.shadow.host.setAttribute("dir", this.htmlDirection);
+        this.shadow.adoptedStyleSheets = [globalStyles, itemStyles];
+        this.shadow.append(node);
+
+        this.keysListeners = [];
+
+        this.updateItem = this.updateItem.bind(this);
+        this.toggleItem = this.toggleItem.bind(this);
+        this.removeItem = this.removeItem.bind(this);
+        this.startEdit = this.startEdit.bind(this);
+        this.stopEdit = this.stopEdit.bind(this);
+        this.cancelEdit = this.cancelEdit.bind(this);
+    }
+
+    update(...args) {
+        [...args].forEach((argument) => {
```

Comment:
It's not clear to me why you're spreading `args`, it's already an array as far as I understand.